src/API/Google/AdsCampaign.php

```diff
@@ -93,91 +104,76 @@ public function get_campaigns(): array {
-
-			$campaign = new Campaign(
-				[
-					'name'                         => $params['name'],
-					'advertising_channel_type'     => AdvertisingChannelType::SHOPPING,
-					'advertising_channel_sub_type' => AdvertisingChannelSubType::SHOPPING_SMART_ADS,
-					'status'                       => CampaignStatus::number( 'enabled' ),
-					'campaign_budget'              => $budget,
-					'maximize_conversion_value'    => new MaximizeConversionValue(),
-					'shopping_setting'             => new ShoppingSetting(
-						[
-							'merchant_id'   => $this->options->get_merchant_id(),
-							'sales_country' => $params['country'],
-						]
-					),
-				]
-			);
-
-			$operation = new CampaignOperation();
-			$operation->setCreate( $campaign );
-			$created_campaign = $this->mutate_campaign( $operation );
-			$campaign_id      = $this->parse_campaign_id( $created_campaign->getResourceName() );
+			$results = ( new AdsCampaignQuery() )
+				->set_client( $this->client, $this->options->get_ads_id() )
+				->where( 'campaign.id', $id )
+				->get_results();
 
-			$this->ads_group->set_up_for_campaign( $created_campaign->getResourceName(), $params['name'] );
+			foreach ( $results->iterateAllElements() as $row ) {
+				return $this->convert_campaign( $row );
+			}
 
-			return [
-				'id'     => $campaign_id,
-				'status' => CampaignStatus::ENABLED,
-			] + $params;
+			return [];
 		} catch ( ApiException $e ) {
 			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
 
-			if ( $this->has_api_exception_error( $e, 'DUPLICATE_CAMPAIGN_NAME' ) ) {
-				throw new Exception(
-					__( 'A campaign with this name already exists', 'google-listings-and-ads' ),
-					$this->map_grpc_code_to_http_status_code( $e )
-				);
-			}
-
 			throw new Exception(
 				/* translators: %s Error message */
-				sprintf( __( 'Error creating campaign: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),
+				sprintf( __( 'Error retrieving campaign: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),
 				$this->map_grpc_code_to_http_status_code( $e )
 			);
 		}
 	}
 
 	/**
-	 * Retrieve a single campaign.
+	 * Create a new campaign.
 	 *
-	 * @param int $id Campaign ID.
+	 * @param array $params Request parameters.
 	 *
 	 * @return array
 	 * @throws Exception When an ApiException is caught.
 	 */
-	public function get_campaign( int $id ): array {
+	public function create_campaign( array $params ): array {
 		try {
-			$results = ( new AdsCampaignQuery() )
-				->set_client( $this->client, $this->options->get_ads_id() )
-				->where( 'campaign.id', $id )
-				->get_results();
+			// Operations must be in a specific order to match the temporary ID's.
+			$operations = array_merge(
+				[ $this->budget->create_operation( $params['name'], $params['amount'] ) ],
+				[ $this->create_operation( $params['name'], $params['country'] ) ],
+				$this->container->get( AdsGroup::class )->create_operations(
+					$this->temporary_resource_name(),
```

Comment:
❔ Would this be using the `temporary_resource_name` method inside `AdsGroup` class or it would use the one defined in this `AdsCampaign` class?

Reply:
It's creating a temporary resource name for the Ad Group. Because we create an Ad Group Ad and a Ad Group Criterion which needs to be linked to the Ad Group (so they all use the same temporary resource number, to link them together).